### PR TITLE
Improvements to differential rotation doc

### PR DESCRIPTION
**Pull request by dstansby**
I just tried to use this guide, which overall was great, but here are some small improvements in my opinion:

- Expand explanation of coordinate frames changing in time, with a specific example using a HGC frame
- Make the point about longitude changing when transforming back to a 'real' frame sooner in the guide - as the key message of this tutorial (in my opinion) it deserves to go before explanations of other ways of doing things (ie. using rotated_time and SkyCoord)
- Reduce the number of words in a couple of places to improve readability

### CODE REVIEW

**Review by wtbarnes — APPROVED** [review]

**Review by ayshih — COMMENTED** [review]
I think it would be better if the first example used something conceptually simpler than HGC, namely HGS.  In fact, you could use Earth's location to illustrate that zero longitude at one time is not zero longitude at a different time.

**Comment by dstansby** [comment]
Updated with HGS, and I think I followed what you suggested?

File: docs/code_ref/coordinates/rotatedsunframe.rst [hunk 1]
```diff
@@ -4,9 +4,28 @@ Differential rotation using coordinate frames
 *********************************************
 
 Normally, coordinates refer to a point in inertial space (relative to the barycenter of the solar system).
-Transforming to a different observation time does not move the point at all, but rather only updates the coordinate representation as needed for the origin and axis orientations at the new observation time.
-However, the Sun both moves translationally in inertial space and rotates about its axis over time.
-Thus, a coordinate that points to a solar feature (e.g., the center of the Sun or an active region on its surface) will not continue to point to the same solar feature at other observation times.
+Transforming to a different observation time does not move the point itself, but if the coordinate frame origin and/or axis change with time, the coordinate representation is updated to account for this change.
+In solar physics an example of a frame that changes with time is the heliographic Stonyhurst (HGS) frame. Its origin moves with the center of the Sun, and its x-axis to follow the Sun-Earth line.
```
Comment (ayshih) [inline docs/code_ref/coordinates/rotatedsunframe.rst:8]:
```suggestion
In solar physics an example of a frame that changes with time is the `~sunpy.coordinates.frames.HeliographicStonyhurst` (HGS) frame.
Its origin moves with the center of the Sun, and its orientation rotates such that the longitude component of Earth is zero at any given time.
```
To be accurate, the HGS x-axis is not exactly aligned with the Sun-Earth line.

File: docs/code_ref/coordinates/rotatedsunframe.rst [hunk 2]
```diff
@@ -4,9 +4,28 @@ Differential rotation using coordinate frames
...
+A coordinate in a HGS frame of reference transformed to a HGS frame defined a day later will have a different longitude::
+
+  >>> from astropy.coordinates import ICRS, SkyCoord
```
Comment (ayshih) [inline docs/code_ref/coordinates/rotatedsunframe.rst:11]:
```suggestion
  >>> from astropy.coordinates import SkyCoord
```

File: docs/code_ref/coordinates/rotatedsunframe.rst [hunk 3]
```diff
@@ -4,9 +4,28 @@ Differential rotation using coordinate frames
...
+  (<Longitude 0. deg>, <Longitude -1.01372559 deg>)
+
+but when transformed to an inertial frame of reference we can see that these two coordinates refer to the same point in space::
+
+  >>> hgs_coord.transform_to('icrs')
+  <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
+       (101.79107615, 26.05004621, 0.99601156)>
+  >>> new_hgs_coord.transform_to('icrs')
+  <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
+        (101.79107615, 26.05004621, 0.99601156)>
```
Comment (ayshih) [inline docs/code_ref/coordinates/rotatedsunframe.rst:28]:
```suggestion
  >>> hgs_coord.transform_to('icrs')
  <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
      (101.79107615, 26.05004621, 0.99601156)>
  >>> new_hgs_coord.transform_to('icrs')
  <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
      (101.79107615, 26.05004621, 0.99601156)>
```

**Review by ayshih — APPROVED** [review]
Minor stuff, otherwise looks fine

**Comment by dstansby** [comment]
Doc build failures look un-related, I think this is good to merge?

**Pull request merged**